src/main/java/core/basesyntax/service/FruitTransactionServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public class FruitTransactionServiceImpl implements FruitTransactionService {
+    private final OperationStrategy operationStrategy;
+
+    public FruitTransactionServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void processTransactions(List<FruitTransaction> transactions) {
+        if (transactions.size() == 0) {
+            throw new RuntimeException("CSV file is empty!");
+        } else {
+            for (FruitTransaction transaction: transactions) {
+                operationStrategy.get(transaction.getOperation()).handle(transaction);
+            }
+        }
```

Comment:
```suggestion
            for (FruitTransaction transaction: transactions) {
                operationStrategy.get(transaction.getOperation()).handle(transaction);
            }
```
Не нужно изобретать бизнес логику. Пустой файл, транзакций нет, склад пустой. Ну и ладно, может в этот день торговли не было. Вот если напишут кидать ошибку при пустом файле - тогда да.